### PR TITLE
feat: Add Live Emote Updates for BTTV

### PR DESCRIPTION
**Pull request by Nerixyz**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Same as 7TV. ~~Waiting on BTTV in night/betterttv#5740 and PubSub test server in Chatterino/twitch-pubsub-server-test#34.~~

### CODE REVIEW

**Comment by pajlada** [comment]
I don't want personal emotes, but it seems like a prerequisite for using their live emote updates.
This PR in the current state doesn't implement personal emotes despite it seeming to be a requirement

**Comment by Felanbird** [comment]
@Nerixyz can you resolve conflicts for this? Makes more sense for you to do it than me since https://github.com/Chatterino/chatterino2/pull/4275 is the culprit 🙂

**Review by Felanbird — APPROVED** [review]
Functionality works as expected 👍 - alias tooltip will come in a followup PR

File: src/providers/bttv/BttvEmotes.hpp [hunk 1]
```diff
@@ -33,6 +35,41 @@ class BttvEmotes final
...
+    /**
+     * Removes an emote from this `map`.
+     * This will _copy_ the emote map and
+     * update the `Atomic`.
+     *
+     * @return The removed emote if any emote was removed.
+     */
+    static boost::optional<EmotePtr> removeEmote(
+        Atomic<std::shared_ptr<const EmoteMap>> &map,
+        const BttvLiveUpdateEmoteRemoveMessage &message);
```
Comment (pajlada) [inline src/providers/bttv/BttvEmotes.hpp:71]: Can the `map` parameter and/or comments be updated to clarify that this function modifies channel specific emotes?

File: src/providers/bttv/BttvEmotes.hpp [hunk 2]
```diff
@@ -33,6 +35,41 @@ class BttvEmotes final
...
+     * Updates an emote in this `map`.
+     * This will _copy_ the emote map and
+     * update the `Atomic`.
+     *
+     * @return pair<old emote, new emote> if any emote was updated.
+     */
+    static boost::optional<std::pair<EmotePtr, EmotePtr>> updateEmote(
+        const QString &channelDisplayName,
+        Atomic<std::shared_ptr<const EmoteMap>> &map,
+        const BttvLiveUpdateEmoteUpdateAddMessage &message);
```
Comment (pajlada) [inline src/providers/bttv/BttvEmotes.hpp:60]: Can the `map` parameter and/or comments be updated to clarify that this function modifies channel specific emotes?

File: src/providers/bttv/BttvEmotes.hpp [hunk 3]
```diff
@@ -33,6 +35,41 @@ class BttvEmotes final
...
+     * Adds an emote to the `map`.
+     * This will _copy_ the emote map and
+     * update the `Atomic`.
+     *
+     * @return The added emote.
+     */
+    static EmotePtr addEmote(
+        const QString &channelDisplayName,
+        Atomic<std::shared_ptr<const EmoteMap>> &map,
+        const BttvLiveUpdateEmoteUpdateAddMessage &message);
```
Comment (pajlada) [inline src/providers/bttv/BttvEmotes.hpp:48]: Can the `map` parameter and/or comments be updated to clarify that this function modifies channel specific emotes?

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -6,7 +6,7 @@ on:
   workflow_dispatch:
 
 env:
-  TWITCH_PUBSUB_SERVER_IMAGE: ghcr.io/chatterino/twitch-pubsub-server-test:v1.0.5
+  TWITCH_PUBSUB_SERVER_IMAGE: ghcr.io/chatterino/twitch-pubsub-server-test:master
```
Comment (pajlada) [inline .github/workflows/test.yml:9]: TODO: UPDATE THIS TO `v1.0.6` WHEN I'VE MADE THIS RELEASE

**Review by pajlada — CHANGES_REQUESTED** [review]
Code quality wise this looks great, thank you for working on this!
I have not tested this yet but will do so tomorrow so I might

**Review by pajlada — APPROVED** [review]

**Review by pajlada — APPROVED** [review]

**Pull request merged**